### ISSUE TITLE
Wrong initial value for onColumnResize if used in combination with grow

### ISSUE THREAD

**Issue by LukasMasuch** [glideapps/glide-data-grid#479]
The new `grow` feature is super useful :) However, there seems to be an issue if it is used together with `onColumnResize`. I would like to deactivate `grow` for a column once it is resized. But the initial width of the column from `onColumnResize` does not correspond to the actual width of the column -> which leads to some jitter (at the beginning of the video):

https://user-images.githubusercontent.com/2852129/186427880-3cfe92b7-c82e-4189-9194-329de4453813.mov

I assume that the initial width does not include the width that is added via the grow feature. But it seems to work fine after the initial rendering. 

Here is a snippet of what I'm trying to do:
```typescript
  const columns = getColumns(element, data).map(column => {
    // Apply column widths from state
    if (
      column.id &&
      columnSizes.has(column.id)
    ) {
      return {
        ...column,
        width: columnSizes.get(column.id),
        grow: 0, // Deactivate grow for this column
      } as GridColumnWithCellTemplate
    }
    return column
  })

  ...

  const onColumnResize = React.useCallback(
    (column: GridColumn, newSize: number) => {
      if (column.id) {
        setColumnSizes(new Map(columnSizes).set(column.id, newSize))
      }
    },
    [columns]
  )
  ...
```

Edit: I just saw [this comment](https://github.com/glideapps/glide-data-grid/pull/406#issue-1277833365) from @jassmith. I assume there isn't any solution for this yet, or? 

Edit 2: I think the issue is more general: when updating a column from using `grow` to `grow==0`, the size is immediately reset to the initial column width.

**Comment by jassmith** [comment]
Okay I understand the problem after spending some time poking around. There is only one good solution here, the column resize events need to give out not just the newSize, but also the newSizeWithGrow. In the fast where you want to retain grow you need the first number, in the case where you wish to do what you are doing, you need the second number. I think... Math is hard and I only bared graduated middle school.

**Comment by jassmith** [comment]
Fixed in 5.0.1

**Comment by LukasMasuch** [comment]
@jassmith Perfect, thanks for the quick fix. This works for me :)
